src/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.Ssl.cs

```diff
@@ -279,4 +290,99 @@ private SafeSslHandle() : base(IntPtr.Zero, true)
         {
         }
     }
+
+    internal sealed class SafeChannelBinding : SafeHandle
+    {
+        [StructLayout(LayoutKind.Sequential)]
+        private struct SEC_CHANNEL_BINDINGS
+        {
+            internal int dwInitiatorAddrType;
+            internal int cbInitiatorLength;
+            internal int dwInitiatorOffset;
+            internal int dwAcceptorAddrType;
+            internal int cbAcceptorLength;
+            internal int dwAcceptorOffset;
+            internal int cbApplicationDataLength;
+            internal int dwApplicationDataOffset;
+        }
+
+        private static readonly byte[] s_tlsServerEndPointByteArray = Encoding.UTF8.GetBytes("tls-server-end-point:");
+        private static readonly byte[] s_tlsUniqueByteArray = Encoding.UTF8.GetBytes("tls-unique:");
+        private static readonly int s_secChannelBindingSize = Marshal.SizeOf<SEC_CHANNEL_BINDINGS>();
+        private readonly int _cbtPrefixByteArraySize;
+        private const int CertHashMaxSize = 128;
+
+        internal int Length
+        {
+            get;
+            private set;
+        }
+
+        internal IntPtr CertHashPtr
+        {
+            get;
+            private set;
+        }
+
+        internal void SetCertHash(byte[] certHashBytes)
+        {
+            Debug.Assert(certHashBytes != null, "check certHashBytes is not null");
+            int length = certHashBytes.Length;
+            Marshal.Copy(certHashBytes, 0, CertHashPtr, length);
+            SetCertHashLength(length);
+        }
+
+        private byte[] GetPrefixBytes(ChannelBindingKind kind)
+        {
+            if (kind == ChannelBindingKind.Endpoint)
+            {
+                return s_tlsServerEndPointByteArray;
+            }
+            else if (kind == ChannelBindingKind.Unique)
+            {
+                return s_tlsUniqueByteArray;
+            }
+            else
+            {
+                throw new NotSupportedException();
+            }
+        }
+
+        internal SafeChannelBinding(ChannelBindingKind kind)
+            : base(IntPtr.Zero, true)
+        {
+            byte[] cbtPrefix = GetPrefixBytes(kind);
+            _cbtPrefixByteArraySize = cbtPrefix.Length;
+            handle = Marshal.AllocHGlobal(s_secChannelBindingSize + _cbtPrefixByteArraySize + CertHashMaxSize);
+            IntPtr cbtPrefixPtr = handle + s_secChannelBindingSize;
+            Marshal.Copy(cbtPrefix, 0, cbtPrefixPtr, _cbtPrefixByteArraySize);
+            CertHashPtr = cbtPrefixPtr + _cbtPrefixByteArraySize;
```

Comment:
Can this just be a managed byte[] that we create with `new byte[cbtPrefix + CertHashMaxSize]`?  Then we can just pass the byte[] instead of IntPtr into the SslGetPeerFinished and SslGetFinished methods.
The native methods would then be defined with `uint8_t* buf` instead of `void* buf`.

Basically, I don't think we need to use native memory here. I think we can get away with keeping this as managed memory.